### PR TITLE
Minimap2: Selfmapping + paf output

### PR DESCRIPTION
**Pull request by willemdek11**
Added the option to map reads against itself.
Added the option for an paf output format.
Added a test for the new options.

### CODE REVIEW

File: tools/minimap2/test-data/mini_reads.fq [hunk 1]
```diff
@@ -0,0 +1 @@
+mini_pacbio_filtered.fastq
```
Comment (shiltemann) [inline tools/minimap2/test-data/mini_reads.fq]: this symlink shouldnt be needed right?

**Comment by shiltemann** [comment]
thanks @willemdek11 

ping @mvdbeek, any comments/suggestions?

File: tools/minimap2/minimap2.xml [hunk 2]
```diff
@@ -146,6 +156,7 @@
                 <option value="paired">Paired</option>
                 <option value="paired_collection">Paired Collection</option>
                 <option value="paired_iv">Paired Interleaved</option>
+                <option value="self_mapping">Reference file</option>
```
Comment (mvdbeek) [inline tools/minimap2/minimap2.xml]: I don't think this is the right place to put this option. I think it'd be more intuitive to place this with the analysis_type_selector selector.

**Comment by mvdbeek** [comment]
Thanks @willemdek11, apart from the placement in the tool form this looks good to me and is a great addition!

File: tools/minimap2/minimap2.xml (hunk 2 above)
Comment (mvdbeek) [reply tools/minimap2/minimap2.xml]: Thought about this some more, and noticed that we don't need to change anything for self_mapping. If you select the same history item as reference and choose the right fastq file you get the desired behaviour. I've opened a PR (https://github.com/willemdek11/tools-iuc/pull/1) against your branch, if you merge this I'm happy to merge this PR.

**Comment by mvdbeek** [comment]
Sorry, I forgot 2 more things: If minimap2 generates an index for the fastq file, will it be stored next to the fastq file ? In that case we have to symlink the file into the working directory. And we probably want to bump the version to `2.5+gx1`

**Comment by willemdek11** [comment]
As far as I can tell is the indexing only saved as the -d option is given. This is option is not (yet) available in the Galaxy tool. So a symlink wouldn't be necessary?

**Comment by mvdbeek** [comment]
OK, cool, so just a bump of the version and I'll merge.

**Comment by mvdbeek** [comment]
Awesome, thanks for the contribution!

**Pull request merged**